src/libpriv/rpmostree-core.c

```diff
@@ -1564,146 +1564,37 @@ gv_nevra_hash (gconstpointer v)
   return g_str_hash (g_variant_get_string (nevra, NULL));
 }
 
-/* We need to make sure that only the pkgs in the base allowed to be
- * removed are removed. The issue is that marking a package for DNF_INSTALL
- * could uninstall a base pkg if it's an update or obsoletes it. There doesn't
- * seem to be a way to tell libsolv to not touch some pkgs in the base layer,
- * so we just inspect its solution in retrospect. libdnf has the concept of
- * protected packages, but it still allows updating protected packages.
- */
 static gboolean
-check_goal_solution (RpmOstreeContext *self,
-                     GPtrArray        *removed_pkgnames,
-                     GPtrArray        *replaced_nevras,
-                     GError          **error)
+collect_replaced_packages (RpmOstreeContext *self,
```

Comment:
IIUC, here it would be helpful a docstring mentioning that this populates `self->pkgs_to_replace` and that it can be called at most once on a given context. (The last part perhaps can be converted from an assert to a more verbose error check?)